### PR TITLE
Migrate wp editor meta box test to Playwright

### PR DESCRIPTION
**Pull request by Rink9**
## What?
Migrate [wp-editor-meta-box.test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/plugins/wp-editor-meta-box.test.js) to its Playwright version.

## Why?
See https://github.com/WordPress/gutenberg/pull/38570 for its background and rationale.

## How?
See [MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.

## Testing Instructions
`npm run test-e2e:playwright test/e2e/specs/editor/plugins/wp-editor-meta-box.spec.js`

## Screencast 

https://user-images.githubusercontent.com/44554636/171704260-fd786681-d02c-4243-90f2-c1bdc1fb66e5.mov

### CODE REVIEW

**Comment by Rink9** [comment]
@ntwb It seems CI are failing here can you please help me by reviewing this PR?

**Comment by Rink9** [comment]
Hi, @kevin940726 Can you please help me with reviewing this PR? It seems CI are failing here, what will be the nest step to move forward? Thanks :)

**Comment by kevin940726** [comment]
Hi @Rink9!

You can click on the "Details" link next to the CI check to see exactly where it fails. For instance, clicking on the "[Details](https://github.com/WordPress/gutenberg/runs/6714289172?check_suite_focus=true#step:5:88)" link of the "Static Analysis" check will show you that it fails on linting the JavaScript code. You can then go and fix that if it's related to your changes.

You can see the [testing guideline](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/testing-overview.md) for more detailed information. GLHF!

**Comment by Rink9** [comment]
@kevin940726 
As I am currently noticing the static analysis CI that is failing is not related to the files I have changed. Feel free to give your feedback regarding this and let me know if I have anything more to do for this PR to get kindly reviewed by you.

image

**Comment by kevin940726** [comment]
@Rink9 Those are only _warnings_ and exist on `trunk` too so it's okay (recommended) to ignore them in this PR :)

The ones that fail the check are the _errors_ in the report. You can scroll to the very bottom of the report and see that there's `1 error`.

**Comment by Rink9** [comment]
Hello @kevin940726, I have solved the problem with failing CI. Would you be able to take a look and review my PR? Please let me know if any further steps are necessary from my end.

**Review by kevin940726 — COMMENTED** [review]
Hi! Have you read the [migration guide](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) and the [best practices](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md)? It's recommended to follow the best practices while migrating the tests so that we can leverage the full power of Playwright. It means that in most cases you have to understand exactly what the tests do (or are trying to do) and modify them accordingly. The most common modifications involve changing to use role selectors and using the locator API. It could be quite difficult in some cases though, feel free to ping us if you have encountered any questions! ;)

**Comment by Rink9** [comment]
Hi, @kevin940726 I addressed your feedbacks regarding `role selectors`. Can you please review that now are we good to go?

**Comment by Rink9** [comment]
@kevin940726 Sorry for pinging you again.... Any updates related to this PR?

**Comment by Rink9** [comment]
Hi @talldan Can you please take e look and review the PR?

File: test/e2e/specs/editor/plugins/wp-editor-meta-box.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-plugin-wp-editor-meta-box'
+		);
+	} );
+
+	test( 'Should save the changes', async ( { admin, editor, page } ) => {
+		await admin.createNewPost();
+		// Add title to enable valid non-empty post save.
+		await page.type( '.editor-post-title__input', 'Hello Meta' );
```
Comment (talldan) [inline test/e2e/specs/editor/plugins/wp-editor-meta-box.spec.js]: It'd be good to change this to a role selector:
```suggestion
		await page.type( 'role=textbox[name="Add title"i]', 'Hello Meta' );
```

File: test/e2e/specs/editor/plugins/wp-editor-meta-box.spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		 * hasWpautop has been errorneously set to false in the line above. You may want to check:
+		 * * Is window.wp.editor.autop a function? It should be one since https://github.com/WordPress/gutenberg/pull/33228
+		 * * Is wpautop still true in the second link mentioned in this comment?
+		 *
+		 * For more context, see https://github.com/WordPress/gutenberg/pull/33228/files#r666897885
+		 */
+		const content = page.locator( '#test_tinymce_id' );
+		expect( await content.evaluate( ( node ) => node.value ) ).toBe(
+			'Typing in a metabox'
+		);
```
Comment (talldan) [inline test/e2e/specs/editor/plugins/wp-editor-meta-box.spec.js]: It should be possible to simplify this in Playwright:
```suggestion
		await expect( content ).toHaveValue( 'Typing in a metabox' );
```

**Review by talldan — COMMENTED** [review]
Thanks for working on this! I left a couple of comments.

One other thing missing is that the old test file (https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/plugins/wp-editor-meta-box.test.js) also needs to be deleted as part of this PR.

**Comment by Rink9** [comment]
Hi @talldan Thanks for your feedback! I have addressed your reviews. Can you please check once?

**Comment by Rink9** [comment]
Hi @talldan It's been a long time! Any updates related to this PR?

**Comment by Rink9** [comment]
Hi @talldan @kevin940726 If anyone can arrange some time, could you please kindly check this PR?

**Comment by talldan** [comment]
I mentioned it in a DM to @Rink9, but mentioning it here as well for transparency. 

I think there's still one thing to address on the PR - it looks like the old test file still needs to be deleted. Although I'm seeing it deleted in 3501674, I still can't see this change in the 'Files Changed' tab for the PR. Not sure why that would be, but hopefully it can be fixed.

**Comment by Rink9** [comment]
Hi @talldan I just have fixed this file changed problem. Can you please take a look and share feedback?

**Review by talldan — APPROVED** [review]
Thanks for all your work here! Congratulations on getting your first PR merged. 🎉

**Comment by Rink9** [comment]
@talldan Thank you so much for your quick response. :blush:

**Pull request merged**